### DISCUSSION TITLE
SendEmail rejected by Email Provider including self-emailing.

### DISCUSSION THREAD

**Discussion by zawhtut** [microsoft/hack-together#158] [Q&A]
Hi...

How can I make sure to send mail using Graph API with app-only authentication?
The app was working initially but then I've been testing it several times. It was kind of blocked or blacklisted I guess.
Do I have to make my app verified app? What could I do at this point?
Thanks for any help...
Sincerely,
...

**Comment by aycabas** [comment]
Hi @zawhtut, you can send an email by using application permissions, here is the reference doc: https://learn.microsoft.com/en-us/graph/api/user-sendmail?view=graph-rest-1.0&tabs=http

If you can share the error you are getting, I can help you further.

**Comment by zawhtut** [comment]
Yes... @aycabas, Thanks for the link... My app can go through the Graph API. It probably was just that email service providers are not trusting the domain from which my emails are originated, or emails need to have extra information such as headers, encoding, or other things that I have yet to learn. I've attached a screenshot of delivery failure emails that didn't make it through. But some go through all the way. 

![Screen Shot 2023-03-15 at 12 32 26 AM](https://user-images.githubusercontent.com/10856644/225283469-c82cdbf1-b2a9-4494-ba40-093213e0569d.png)
![Screen Shot 2023-03-15 at 12 32 42 AM](https://user-images.githubusercontent.com/10856644/225283553-d2444813-8293-4da4-944c-6c86c3563d69.png)

**Reply by waldekmastykarz** [reply]
Who is the sender of the message? Could it be that some information is missing from the message which is why it's being rejected?

**Comment by zawhtut** [comment]
Thanks @waldekmastykarz for reaching out to me... 
I am just sending emails to myself using the same email address that I signed up with AAD developer subscription... 
So... With that initial problem, I looked up the email headers and analyze the email header. 

![Screen Shot 2023-03-15 at 2 39 26 PM](https://user-images.githubusercontent.com/10856644/225454386-5e3e74c8-d193-4895-983c-188291c05b75.png)

I found that `Authentication-Results: dkim=none (message not signed) header.d=none;dmarc=none action=none header.from=2k2bql.onmicrosoft.com;` DKIM is not signed.

And then I look up about DKIM found this to configure...
https://learn.microsoft.com/en-us/microsoft-365/security/office-365-security/email-authentication-dkim-configure

![Screen Shot 2023-03-15 at 10 51 36 AM](https://user-images.githubusercontent.com/10856644/225459927-205f72fe-2357-4757-8f1a-73dd16abb648.png)

Even though I enable the DKIM in that security settings, DKIM in the message headers are not signed yet.
And I couldn't configure the DKIM manually because I can't use power shell to set the configuration. I can't get the power shell because I can only be on 365 developer subscription, and can't sign up Azure subscription with that account.

Anyhow I found another settings in Exchange Online. So I switch my domain to `InternalRelay` type. 
So far it is working that my messages from Blazor app using Microsoft Graph are delivered to my emails...

![Screen Shot 2023-03-15 at 3 30 52 PM](https://user-images.githubusercontent.com/10856644/225458935-4312d168-b2e9-4dcd-a0fe-0784f38f21fb.png)


CC: @aycabas
